crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -84,33 +86,49 @@ impl GovernanceReadApi {
             self.get_system_state()?.into_sui_system_state_summary();
         let mut delegated_stakes = vec![];
         for ((pool_id, validator_address), stakes) in pools {
+            // Rate table and rate can be null when the pool is not active
             let rate_table = self
                 .get_exchange_rate_table(&system_state, &pool_id)
-                .await?;
-
-            let current_rate = self
-                .get_exchange_rate(rate_table, system_state.epoch)
-                .await?;
+                .await
+                .ok();
+            let current_rate = if let Some(rate_table) = rate_table {
+                self.get_exchange_rate(rate_table, system_state.epoch)
+                    .await
+                    .ok()
+            } else {
+                None
+            };
 
             let mut delegations = vec![];
             for stake in stakes {
                 // delegation will be active in next epoch
-                let status = if system_state.epoch >= stake.request_epoch() {
-                    let stake_rate = self
-                        .get_exchange_rate(rate_table, stake.request_epoch())
-                        .await?;
-                    let estimated_reward = (((stake_rate.rate() / current_rate.rate()) - 1.0)
-                        * stake.principal() as f64)
-                        .round() as u64;
+                let status = if system_state.epoch >= stake.activation_epoch() {
+                    let estimated_reward = if let (Some(rate_table), Some(current_rate)) =
+                        (&rate_table, &current_rate)
+                    {
+                        let stake_rate = self
+                            .get_exchange_rate(*rate_table, stake.activation_epoch())
+                            .await
+                            .unwrap_or_default();
+                        let estimated_reward = ((stake_rate.rate() / current_rate.rate()) - 1.0)
+                            * stake.principal() as f64;
+                        if estimated_reward < 0.0 {
```

Comment:
We can probably do something like `max(0.0, estimated_reward).round() as u64` here.